core/network/src/service.rs

```diff
@@ -521,136 +519,83 @@ impl<B: BlockT, F: FnOnce(&mut ConsensusGossip<B>, &mut Context<B>)> GossipTask<
-			protocol,
-			service_clone,
-			peers,
-			import_queue,
-			transaction_pool,
-			finality_proof_provider,
-			network_port,
-			protocol_rx,
-			status_sinks,
-			peerset_clone,
-			on_demand_in
-		)
-			.select(close_rx.then(|_| Ok(())))
-			.map(|(val, _)| val)
-			.map_err(|(err,_ )| err);
-
-		// Note that we use `block_on` and not `block_on_all` because we want to kill the thread
-		// instantly if `close_rx` receives something.
-		match runtime.block_on(fut) {
-			Ok(()) => debug!(target: "sub-libp2p", "Networking thread finished"),
-			Err(err) => error!(target: "sub-libp2p", "Error while running libp2p: {:?}", err),
-		};
-	})?;
 
-	Ok(((close_tx, thread), service, peerset))
+	/// Interval at which we send status updates on the `status_sinks`.
+	status_interval: tokio_timer::Interval,
+	/// Interval at which we update the `connected_peers` Arc.
+	connected_peers_interval: tokio_timer::Interval,
 }
 
-/// Runs the background thread that handles the networking.
-fn run_thread<B: BlockT + 'static, S: NetworkSpecialization<B>, H: ExHashT>(
-	is_offline: Arc<AtomicBool>,
-	is_major_syncing: Arc<AtomicBool>,
-	mut protocol: Protocol<B, S, H>,
-	network_service: Arc<Mutex<NetworkService<Message<B>>>>,
-	peers: Arc<RwLock<HashMap<PeerId, ConnectedPeer<B>>>>,
-	import_queue: Box<ImportQueue<B>>,
-	transaction_pool: Arc<dyn TransactionPool<H, B>>,
-	finality_proof_provider: Option<Arc<FinalityProofProvider<B>>>,
-	mut network_port: mpsc::UnboundedReceiver<NetworkMsg<B>>,
-	mut protocol_rx: mpsc::UnboundedReceiver<ProtocolMsg<B, S>>,
-	status_sinks: Arc<Mutex<Vec<mpsc::UnboundedSender<ProtocolStatus<B>>>>>,
-	peerset: PeersetHandle,
-	mut on_demand_in: Option<mpsc::UnboundedReceiver<RequestData<B>>>,
-) -> impl Future<Item = (), Error = io::Error> {
-	// Implementation of `protocol::NetworkOut` using the available local variables.
-	struct Ctxt<'a, B: BlockT>(&'a mut NetworkService<Message<B>>, &'a PeersetHandle);
-	impl<'a, B: BlockT> NetworkOut<B> for Ctxt<'a, B> {
-		fn report_peer(&mut self, who: PeerId, reputation: i32) {
-			self.1.report_peer(who, reputation)
-		}
-		fn disconnect_peer(&mut self, who: PeerId) {
-			self.0.drop_node(&who)
-		}
-		fn send_message(&mut self, who: PeerId, message: Message<B>) {
-			self.0.send_custom_message(&who, message)
-		}
-	}
+impl<B: BlockT + 'static, S: NetworkSpecialization<B>, H: ExHashT> Future for NetworkMut<B, S, H> {
+	type Item = ();
+	type Error = io::Error;
 
-	// Interval at which we send status updates on the `status_sinks`.
-	let mut status_interval = tokio::timer::Interval::new_interval(STATUS_INTERVAL);
-	// Interval at which we update the `connected_peers` Arc.
-	let mut connected_peers_interval = tokio::timer::Interval::new_interval(CONNECTED_PEERS_INTERVAL);
+	fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+		// Implementation of `protocol::NetworkOut` using the available local variables.
+		struct Ctxt<'a, B: BlockT>(&'a mut NetworkService<Message<B>>, &'a PeersetHandle);
```

Comment:
This is an old code, but now is a good time to rename it to something readable according to our guidelines, like "Context"